src/_guides/language/coming-from/js-to-dart.md

```diff
@@ -0,0 +1,2835 @@
+---
+title: Learning Dart as a JavaScript developer
+description: Leverage your JavaScript knowledge when learning Dart.
+---
+
+This guide aims to leverage your JavaScript (JS)
+programming knowledge when learning Dart.
+It showcases key similarities and differences
+in both languages, and introduces Dart concepts
+that are not present in vanilla JavaScript.
+As a JavaScript developer,
+Dart should feel quite familiar,
+as both languages share many concepts. 
+
+Like JavaScript, Dart runs on an event loop,
+so both languages execute code in a similar way.
+For example, asynchronous concepts like `Futures`
+(`Promises` in JS) and the `async/await` syntax are very similar.
+
+Dart is strongly typed, unlike JavaScript.
+If you are also familiar with TypeScript or tooling
+that adds static typing (like Flow), then learning Dart
+probably won't be much of an extra step for you.
+However, if you've mostly worked with vanilla JavaScript,
+it might take a bit more effort to adjust.
+The good news is that since Dart is strongly typed,
+many errors that might exist in JavaScript code are
+caught even before compiling your Dart code.
+
+As of Dart 2.0, null safety is enabled by default,
+which JavaScript doesn't support. As a JavaScript developer,
+it might take a while to learn how to write null safe code,
+but the trade-off is better protection against
+null reference exceptions that are detected even
+before compiling Dart code. (Thereby avoiding those
+dreaded `TypeErrors` that occur when doing operations
+on a JS variable that turns out to be null.)
+
+## Conventions and linting
+
+JavaScript and Dart both have linting tools
+to enforce standard conventions. However,
+while JavaScript has `ESLint` as a standalone tool
+(among other available tooling),
+and many different standards and configurations,
+Dart has official layout conventions and includes
+a linter to make compliance effortless.
+To customize the lint rules for your project,
+follow the [Customizing static analysis][]
+instructions on dart.dev.
+
+{{site.alert.secondary}}
+  **Pro tip:** Dart provides [`dart fix`][],
+  which finds and fixes errors found by the analyzer.
+{{site.alert.end}}
+
+Dart also provides a code formatter,
+which is similar to JS tools like [Prettier][].
+Automatically format any Dart project by running
+`dart format` on the command line. For Flutter,
+`flutter format` acts as an alias for this command.
```

Comment:
Maybe we can link to the `dart format` documentation here like we do for `dart fix` above:

```suggestion
[`dart format`](/tools/dart-format) on the command line. 
For Flutter, `flutter format` acts as an alias for this command.
```